test/avrora/storage/registry_test.exs

```diff
@@ -161,6 +161,84 @@ defmodule Avrora.Storage.RegistryTest do
 
       Application.put_env(:avrora, :registry_url, registry_url)
     end
+
+    test "when registry auth is malformed" do
+      registry_auth = Application.get_env(:avrora, :registry_auth)
+      Application.put_env(:avrora, :registry_auth, {:basic, "user", "pass", "unknown"})
```

Comment:
To avoid this get-set-reset way in tests I was [using Mox](https://github.com/Strech/avrora/blob/master/test/support/mocks.ex). It allows writing `MyMock.expect(...)`.

I think, maybe it makes sense to move to the same way, but with `Config`, it might be a bit overkill, but will allow per test-case settings.